l10n_it_edi_extension/models/account_move.py

```diff
@@ -0,0 +1,874 @@
+                    {
+                        "l10n_it_edi_related_invoice_code": get_text(
+                            element_parent_invoice, ".//NumeroFatturaPrincipale"
+                        ),
+                        "l10n_it_edi_related_invoice_date": get_date(
+                            element_parent_invoice, ".//DataFatturaPrincipale"
+                        ),
+                    }
+                )
+
+        tag_name = (
+            ".//DettaglioLinee"
+            if not extra_info["simplified"]
+            else ".//DatiBeniServizi"
+        )
+        if elements_line := body_tree.xpath(tag_name):
+            for element_line in elements_line:
+                self.l10n_it_edi_amount_untaxed += get_float(
+                    element_line, ".//PrezzoTotale"
+                )
+
+        if elements_summary := body_tree.xpath(".//DatiBeniServizi/DatiRiepilogo"):
+            for element_summary in elements_summary:
+                self.env["l10n_it_edi.summary_data"].create(
+                    {
+                        "tax_rate": get_float(element_summary, ".//AliquotaIVA"),
+                        "non_taxable_nature": get_text(element_summary, ".//Natura"),
+                        "incidental_charges": get_float(
+                            element_summary, ".//SpeseAccessorie"
+                        ),
+                        "rounding": get_float(element_summary, ".//Arrotondamento"),
+                        "amount_untaxed": get_float(
+                            element_summary, ".//ImponibileImporto"
+                        ),
+                        "amount_tax": get_float(element_summary, ".//Imposta"),
+                        "payability": get_text(element_summary, ".//EsigibilitaIVA"),
+                        "law_reference": get_text(
+                            element_summary, ".//RiferimentoNormativo"
+                        ),
+                        "invoice_id": self.id,
+                    }
+                )
+                self.l10n_it_edi_amount_tax += get_float(element_summary, ".//Imposta")
+
+        extra_info["l10n_it_edi_ext_body_tree"] = body_tree
+        return extra_info, message_to_log
+
+    def _l10n_it_edi_create_partner(
+        self, xml_tree, partner_section_xpath, vat, codice_fiscale
+    ):
+        country_id = False
+        is_company = bool(get_text(xml_tree, partner_section_xpath + "//Denominazione"))
+        eori_code = get_text(xml_tree, partner_section_xpath + "//CodEORI")
+
+        if country_code := get_text(xml_tree, partner_section_xpath + "//IdPaese"):
+            countries = self.env["res.country"].search([("code", "=", country_code)])
+            if countries:
+                country_id = fields.first(countries).id
+            else:
+                raise UserError(
+                    self.env._("Country Code %s not found in system.") % country_code
+                )
+
+        vals = {
+            "vat": country_code + vat,
```

Comment:
Qui chiederei un parere di @TheMule71 che lo so che gli piace l'argomento :wink:

Reply:
È possibile salvare nella P.IVA del partner il numero della P.IVA senza il prefisso, quindi possiamo fare sempre così. Quello che troviamo in `IdCodice` sarà scritto in `vat`

Reply:
per @monen17

Reply:
Aggiunto in https://github.com/Borruso/l10n-italy/pull/33